### ISSUE TITLE
Meta-Learner Reproducibility

### ISSUE THREAD

**Issue by imaginarymuffin** [uber/causalml#532]
**Description**
On first attempt, meta-learners perform great: lift and gain curves look reasonable and I can identify groups with stark heterogeneous effects. 
Whenever the notebook is run again, cannot to replicate the results even by trying several random_state and seed combinations. 
If these heterogeneous effects groups _do exist_ in the data, why is it so hard to identify them again?

**Code**

`X=df_reg[features_fullrank]
y=df_reg[target_var]
treatment=df_reg[treatment]`

`X_train, X_test, y_train, y_test,  treatment_train, treatment_test  =  train_test_split(X, y,  treatment, test_size=0.5, random_state=42)`

`learner_x = BaseRRegressor(learner=XGBRegressor(random_state=10))`

`learner_x.fit(X=X_train, treatment=treatment_train, y=y_train)`

**Screenshots**
**First Attempt**

<img width="586" alt="Screen Shot 2022-07-14 at 8 55 57 AM" src="https://user-images.githubusercontent.com/105323039/179054204-cf60c777-15eb-454b-9b12-8b8bdd1deeb2.png">
<img width="580" alt="Screen Shot 2022-07-14 at 8 56 02 AM" src="https://user-images.githubusercontent.com/105323039/179054092-eb690fb5-3ed3-4733-a294-c853dd2be735.png">
<img width="779" alt="Screen Shot 2022-07-14 at 8 56 19 AM" src="https://user-images.githubusercontent.com/105323039/179054230-8ea5a7f5-512f-4207-ad94-9ffa48adb989.png">

**Any further attempt**
<img width="603" alt="Screen Shot 2022-07-14 at 8 57 39 AM" src="https://user-images.githubusercontent.com/105323039/179054256-e7384fc0-e699-42e8-b870-62c8c5f7b12e.png">
<img width="668" alt="Screen Shot 2022-07-14 at 8 57 44 AM" src="https://user-images.githubusercontent.com/105323039/179054681-f3cfbf0a-5bbc-4c06-a5a7-ef0c53282000.png">
<img width="770" alt="Screen Shot 2022-07-14 at 8 58 00 AM" src="https://user-images.githubusercontent.com/105323039/179054710-e3a536b3-e569-4f55-9d08-e33a6fdd5d3b.png">

**Environment:**
 - OS: Linux-4.14.174-x86_64-with-debian-8.11
 - Python Version: 3.6
 - `pandas==0.25.3`, `scikit-learn==0.22`, `causalml==0.9.0`, `xgboost=0.81`

**Additional context**
I did learn that reproducibility can be ensured by setting `np.random.seed(seed)`, still I am unable to replicate my initial results.

**Comment by ppstacy** [comment]
Hi @imaginarymuffin R-learner itself uses cross-validaiton as well as described [here](https://causalml.readthedocs.io/en/latest/methodology.html#r-learner). I saw you already fix the random_state in `BaseRRegressor(learner=XGBRegressor(random_state=10))`, but R-learner also need the similar input (see code [here](https://github.com/uber/causalml/blob/1fba16ab292fbcea832a021e2ef7729a4cbe4cb9/causalml/inference/meta/rlearner.py#L41)) which said you could try `BaseRRegressor(learner=XGBRegressor(random_state=10), random_state=10)`.
